common/meta/types/src/user_info.rs

```diff
@@ -113,6 +145,14 @@ impl UserOption {
         self.flags.insert(flag);
     }
 
+    pub fn switch_option_flag(&mut self, flag: UserOptionFlag, on: bool) {
```

Comment:
I don't like this API.

How about `set_option_flag(&mut self, flag: UserOptionFlag)`?

For remove, users can call `unset_option_flag` instead.

Or, better naming:

- insert_option_flag
- remove_option_flag

Reply:
without this api, the code in `UserOptionItem` would from:

```
impl UserOptionItem {
    pub fn apply(&self, option: &mut UserOption) {
        match self {
            Self::TenantSetting(enabled) => {
                option.switch_option_flag(UserOptionFlag::TenantSetting, *enabled);
            }
            Self::ConfigReload(enabled) => {
                option.switch_option_flag(UserOptionFlag::ConfigReload, *enabled);
            }
            Self::DefaultRole(v) => option.set_default_role(Some(v.clone())),
        }
    }
}
```

to

```
impl UserOptionItem {
    pub fn apply(&self, option: &mut UserOption) {
        match self {
            Self::TenantSetting(enabled) => {
                if enabled {
                    option.set_option_flag(UserOptionFlag::TenantSetting);
                } else {
                    option.unset_option_flag(UserOptionFlag::TenantSetting);
                }
            }
            Self::ConfigReload(enabled) => {
                if enabled {
                    option.set_option_flag(UserOptionFlag::ConfigReload);
                } else {
                    option.unset_option_flag(UserOptionFlag::ConfigReload);
                }
            }
            Self::DefaultRole(v) => option.set_default_role(Some(v.clone())),
        }
    }
}
```

in my understanding these flag equals setting a boolean value to a struct.

introducing two apis may introduce some duplications in the code.

Reply:
Why do we need to store a `flag` in UserOption? Code could be must simple:

```rust
if enable_tenant_setting {
   option.tenant_setting = true
}
if enable_config_reload {
    option.config_reload = true
}
...
```

Reply:
like this，lemme hide the flag into internal

Reply:
oh, I failed on this refactor due to:

1. there're code directly depends on the flag in the privilege check case 
likes in https://github.com/datafuselabs/databend/blob/59ab2530da692de17cd50864f1843d51d85f2799/query/src/procedures/procedure_factory.rs#L42
2. the flags is persisted in metasrv in binary, can not easily change the storage about flags

this refactor may need more works than I expected, I guess I need a seperate PR to consider this :(